### PR TITLE
Refactoring rasa train

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
- Refactoring of `rasa/train.py` (functionality stayed the same)

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: rasa/train.py [hunk 1]
```diff
@@ -396,28 +403,23 @@ def _train_nlu_with_validated_data(
...
+            uncompress=uncompress,
         )
 
-        return output_path
-
     return _train_path
 
 
-def enrich_config(config_path, missing_keys, FALLBACK_CONFIG_PATH):
+def _enrich_config(config_path, missing_keys, FALLBACK_CONFIG_PATH):
```
Comment (wochinge) [inline rasa/train.py]: let's add types

File: rasa/train.py [hunk 2]
```diff
@@ -440,6 +442,53 @@ def get_valid_config(config: Text, mandatory_keys: List[Text]) -> Text:
             "{}. Filling missing parameters from fallback configuration file: '{}'."
             "".format(config, ", ".join(missing_keys), FALLBACK_CONFIG_PATH)
         )
-        enrich_config(config_path, missing_keys, FALLBACK_CONFIG_PATH)
+        _enrich_config(config_path, missing_keys, FALLBACK_CONFIG_PATH)
 
     return config_path
+
+
+def _package_model(
```
Comment (wochinge) [inline rasa/train.py]: should we move this to `model.py` and maybe write tests?

File: rasa/train.py [hunk 3]
```diff
@@ -440,6 +442,53 @@ def get_valid_config(config: Text, mandatory_keys: List[Text]) -> Text:
...
+    print_success(
+        "Your Rasa model is trained and saved at '{}'.".format(
+            os.path.abspath(output_path)
+        )
+    )
+
+    return output_path
+
+
+def _should_retrain(new_fingerprint, old_model, train_path):
```
Comment (wochinge) [inline rasa/train.py]: same: should we move this to `model.py` and write tests?

File: rasa/train.py [hunk 4]
```diff
@@ -132,18 +128,54 @@ def train(
...
+        return _package_model(
+            new_fingerprint=new_fingerprint,
+            output_path=output_path,
+            train_path=train_path,
+            fixed_model_name=fixed_model_name,
+            uncompress=uncompress,
+        )
+
+    print_success(
+        "Nothing changed. You can use the old model stored at '{}'"
```
Comment (wochinge) [inline rasa/train.py]:
```suggestion
        "Nothing changed. You can use the old model stored at '{}'."
```

File: rasa/train.py [hunk 5]
```diff
@@ -77,12 +77,8 @@ def train(
...
     """
-    config = get_valid_config(config, CONFIG_MANDATORY_KEYS)
-
+    config = _get_valid_config(config, CONFIG_MANDATORY_KEYS)
     train_path = tempfile.mkdtemp()
-    old_model = model.get_latest_model(output_path)
-    retrain_core = True
-    retrain_nlu = True
 
     skill_imports = SkillSelector.load(config)
```
Comment (wochinge) [inline rasa/train.py]: Should we move lines 80 to 95 to a function `_get_training_data`? Or is that a overkill?

Comment (tabergma) [reply rasa/train.py]: As it is not just getting training data but many different kind of things that are needed later on, I would keep it as it is.

File: rasa/train.py (hunk 2 above)
Comment (tabergma) [reply rasa/train.py]: I would like to keep the function in `train.py` as it has a `print_sucess` statement and its functionality is only used in this file. But will write some tests.

**Pull request merged**